lua/terrortown/menus/gamemode/administration/playermodels.lua

```diff
@@ -63,13 +63,17 @@ function CLGAMEMODESUBMENU:Populate(parent)
 			end
 		}, base)
 
-		playermodels.IsSelectedModel(name, function(value)
-			boxCache[name]:SetModelSelected(value, false)
-		end)
-
-		playermodels.IsHattableModel(name, function(value)
-			boxCache[name]:SetModelHattable(value, false)
-		end)
+		-- The anonymous function internally checks if boxCache[name] exists.
+		local outerBox = boxCache[name]
+		if outerBox then
+			playermodels.IsSelectedModel(name, function(value)
+				boxCache[name]:SetModelSelected(value, false)
+			end)
+
+			playermodels.IsHattableModel(name, function(value)
+				boxCache[name]:SetModelHattable(value, false)
+			end)
+		end
```

Comment:
```suggestion
		playermodels.IsSelectedModel(name, function(value)
			if not boxCache[name] then return end
			
			boxCache[name]:SetModelSelected(value, false)
		end)

		playermodels.IsHattableModel(name, function(value)
			if not boxCache[name] then return end
			
			boxCache[name]:SetModelHattable(value, false)
		end)
```

What I meant was around the call of the function you dont need to check it. But inside the anonymous function you do. Because that function may be called later when the f1 menu is not valid anymore for example due to server lag or any other causes.